ftw/simplelayout/browser/anchors.py

```diff
@@ -12,7 +19,36 @@ def listAnchorNames(self, *args, **kwargs):
         query = {'object_provides': ISimplelayoutBlock.__identifier__}
 
         for block in self.context.listFolderContents(contentFilter=query):
-            view = block.restrictedTraverse('content_anchors')
-            anchors.extend(view.listAnchorNames(*args, **kwargs))
+            fields = self.get_html_fields(block)
+
+            for field in fields:
+                anchors.extend(
+                    self.extract_anchors(
+                        field.get(block).output
+                    )
+                )
 
         return anchors
+
+    def get_html_fields(self, obj):
+        fields = []
+        for schema in iterSchemata(obj):
+            for name, field in getFieldsInOrder(schema):
+                # Only consider RichText fields.
+                if IRichText.providedBy(field):
+                    fields.append(field)
+
+        # Filter empty fields (i.e. not having a value).
+        fields = filter(None, fields)
```

Comment:
You are absolutely right. I have improved the code and added a test using an empty text block: https://github.com/4teamwork/ftw.simplelayout/pull/435/files#diff-96374b799c0ac7345fe9ba54e3d3f757R34